MapboxDirections/MBDirectionOptions.swift

```diff
@@ -301,7 +304,10 @@ open class DirectionOptions: NSObject, NSSecureCoding, NSCopying {
         return waypoints.map { "\($0.coordinate.longitude),\($0.coordinate.latitude)" }
     }
     
-    internal var path: String { return "" }
+    internal var path: String {
+        assert(self.path.isEmpty, "path should be overriden by subclass")
```

Comment:
This would result in infinite recursion rather than a failing assertion. Assert `false` instead.